cEP-0022.md

```diff
@@ -0,0 +1,162 @@
+
+| Metadata |                                               |
+| -------- | --------------------------------------------- |
+| cEP      | 22                                            |
+| Version  | 1.0                                           |
+| Title    | Quickstart Green Mode                         |
+| Authors  | Ishan Srivastava <mailto:ishan.srt@gmail.com> |
+| Status   | Implementation Due                            |
+| Type     | Feature                                       |
+
+# Abstract
+
+This document describes what is the **Green Mode** for `coala-quickstart` and how
+will this be implemented.
+
+This cEP describes new type of config files which will be generated by
+`coala-quickstart`, how will they be different from the ones currently being
+created, how is this helpful and the action-plan for adding this feature.
+
+# Introduction
+
+Currently `coala-quickstart` generates config files which point out too many
+errors in the code-base of the project. It also asks too many questions
+while generating the config file. The config files are not that specific to
+the project and the errors pointed out by coala using generated config files
+are generally undesirable for the project owners.
+
+Project owners need a configuration file which *agrees* with their codestyle, is
+as specific as possible i.e. can point out maximum errors with least amount
+of manual edits in the future, for future inconsistencies in commits.
+Many organizations don't follow specific code styles like `PEP8`, instead they
+follow their own standards. We need to adapt to those standards with least
+user interaction and produce config files which generate absolutely no errors
+in their codebase. This will lead to easier adoption of coala across communities
+as it is going to result in an easy mergeable Pull Request regarding
+*adding `.coafile` and CI to enforce it* in the other orgs for example
+<https://github.com/processing/processing/pull/4781>.
+
+# Implementation
+
+This project will be implemented in two phases:
+
+- The Brute Force way
+- Further optimization on the Brute Force
+
+1. The first step will consist of preparing a dict of all the values that can
+   be attained by the bears' settings, may it be necessary or optional as we need
+   to create *green* config files that are as specific as possible. Failure to do
+   so will lead to dropping of most **optional** bear settings, thus will lead to
+   coala not detecting many inconsistencies, and the developers of the project will
+   have to do the work of learning about the bears and their settings and then
+   modifying the `.coafile` manually, while not able to guess the correct
+   **necessary** bear settings will lead to dropping of bears themselves.
+
+**The following steps mainly describe the brute force way**
+
+2. Certain thresholds may be defined for an org based on several factors
+   for the number of inconsistencies detected to be considered mistakes however
```

Comment:
For simplicity, can we move this threshold concept to "Further optimization of brute force section"?

Reply:
Also, giving an example of the factors deciding the threshold, and how would they define the threshold would help.

Reply:
its gonna be done by us until a very long time I presume

Reply:
so maybe based on stars and reputation of the org

Reply:
Ah, I see it's being used in the later steps too. Let's keep it here for now :+1: